### DISCUSSION TITLE
Raspberry Matic findet RFUSB Stick nur in der Web UI und nicht im Home Assistent Addon

### DISCUSSION THREAD

**Discussion by dennisxmeyer** [OpenCCU/OpenCCU#2641] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen, 
ich bin neu hier und freue mich euch kennen zu lernen! 
Nun zu meinem Problem:
**Der RFUSB Stick (original nicht Telekom) wird nur in der WebUI Oberfläche gefunden, im Homessistent Addon leider nicht.**
Dort konnte ich auch Geräte anlernen.
Das bringt mir aber nichts, wenn ich sie im Homeassistent nicht bedienen kann.
In dem RespberryMatic Addon auf dem Homeassistent leider nicht.
Ich versuche seit Tagen alles, aber finde einfach den Fehler nicht.

Ich hab eine Unifi Dreammachine und ein Dritec Vigor Modem.
Ich habe auf meiner Synologie NAS über den Virtual Machine Manager 2 Virtuelle Maschinen erstellt.
1. Home Assistant installiert
2. Raspberry Matic Addon installiert

Hier ein paar Bilder des Systems:
![Addon](https://github.com/jens-maus/RaspberryMatic/assets/157129117/a0fcdb52-d3c1-4a7a-af4c-deec4bddb5b9)
![Nas](https://github.com/jens-maus/RaspberryMatic/assets/157129117/2e44409e-722b-438e-b0d1-2a79c6f1ddc1)
![unifi](https://github.com/jens-maus/RaspberryMatic/assets/157129117/c4403a3f-01fd-49aa-bee2-cfb4ae86505b)
![WebUi](https://github.com/jens-maus/RaspberryMatic/assets/157129117/3f779802-74b1-4ad4-88fb-56ac8f5449a4)

Ich hoffe es kann mir jemand helfen, ich würde wirklich gerne in Zukunft meine Homematic Ip Geräte in Homeassistent steuern können, damit ich mir weitere Shellys etc zulegen kann.

**Comment by jens-maus** [comment, marked as answer]
Die Installation des RaspberryMatic HA Addons ist nur die erste Aktion die notwendig ist um von HA aus eine CCU-Konfigurationsoberfläche für alle HomeMatic/homematicIP Geräte zu haben. Wenn man diese Geräte dann jedoch innerhalb von HA direkt bedienen/auswerten/steuern möchte wie andere nicht-HomeMatic Geräte braucht man noch die passende "Integration". In dem Falle ist das die "Homematic(IP) Local" genannte Integration für die hier die passende Installationsanleitung zu finden ist:

https://github.com/jens-maus/RaspberryMatic/wiki/HomeAssistant-Integration

**Comment by Baxxy13** [comment]
Letztlich laufen hier wohl 2 RaspberryMatic's, also eine zuviel.
Einmal als VM und einmal als HA-AddOn. 

Ich würde das AddOn wieder löschen und dann die VM integrieren.

**Reply by dennisxmeyer** [reply]
Danke, das könnte das Problem sein! 

Ich bin ehrlich gesagt etwas verwirrt, was das angeht. Deswegen sind die dann bestimmt auch doppelt...
Vielleicht kannst du mir noch etwas dabei helfen!?!

Ich habe auf der Synology 2 virtuelle Maschinen.
1. Home Assistent
2. Raspberry Matic

Im Homeassistent das Raspberry Addon und die Homematic(IP) Local Integration

Zustand siehe Bilder

Synology  virtual Maschine Manager:
![1](https://github.com/jens-maus/RaspberryMatic/assets/157129117/bf4daeab-4c10-48bc-b043-c720c3510f23)


Home Assistent Addons:
![2](https://github.com/jens-maus/RaspberryMatic/assets/157129117/7ec6f58d-fe9a-4c5d-91de-ffa2659f4b69)

Home Assistent Integrationen:
![3](https://github.com/jens-maus/RaspberryMatic/assets/157129117/2b93e352-8e7a-4d5d-aaa2-b759d7f96b28)


Das heißt ich soll jetzt das Raspberry Matic Addon im Homeassistent löschen?
Und ich muss dann die Geräte im Raspberry, den ich nur unter der IP Adresse 192.168.1.118 erreiche anlernen?

**Reply by jens-maus** [reply]
> Das heißt ich soll jetzt das Raspberry Matic Addon im Homeassistent löschen? Und ich muss dann die Geräte im Raspberry, den ich nur unter der IP Adresse 192.168.1.118 erreiche anlernen?

Genau so ist es. Frei dem Motto "Es kann nur Einen geben!" musst du dich eben entscheiden: Entweder die RaspberryMatic Zentrale innerhalb der Synology parallel zu HomeAssistant als unabhängiges Betriebssystem betreiben oder du betreibst RaspberryMatic als teils eingeschränktes HA Addon innerhalb von HomeAssistant selbst (d.h. eine virtuelle RaspberryMatic/CCUZentrale innerhalb eines virtuellen HomeAssistant).

Und da tiefe Virtualisierungsebenen nie gut oder von größerem Vorteil sind würde ich dir immer zu ersterem raten. D.h. Lösche das RaspberryMatic Addon innerhalb deines virtuellen HomeAssistant und nutze die "HomematicIP (Local)" integration um zu deiner virtuellen RaspberryMatic die parallel zu HomeAssistant läuft (hier IP 192.168.1.118) die Geräte abzugreifen.

**Reply by dennisxmeyer** [reply]
Vielen Dank!

Das hab ich jetzt gemacht. Dann will ich mal gucken, wie ich der neuen Welt (Home Assistent) klar komme ;-)

Jetzt habe ich noch ein letztes Problem: 
Ich habe 2 Geräte am laufen: **RF-USB Stick** und die **originale Homematic IP Zentrale (CCU)**
Diese werden wir auch beide in HomeAssistent angezeigt.
siehe Bild 
![4](https://github.com/jens-maus/RaspberryMatic/assets/157129117/1e8ec045-96f9-4fd1-ac78-23d197f4b1b7)


Welche der beiden CCU´s (original oder Stick) soll ich jetzt am besten benutzen?

Kann ich die originale CCU weiter laufen lassen und diese sowohl aus Home Assistent, als auch in der Homematic IP APP bedienen bzw. zumindest schalten oder muss ich mich für ein System entscheiden?
Dann bräuchte ich den RF-USB Stick nicht. 
Der Standort der originalen CCU ist auch deutlich besser. Da bin ich ja mit dem Netzwerkkabel flexibel. 
Den Stick muss ich ja per USB anschließen, da bin ich in der letzten Ecke des Hauses und müsste mit Reichweitenverlängerungen über mehrere Funksteckdosen arbeiten.
Jedoch hab ich gelesen, dass es bei der originalen Zentrale wegen zu vielen Aufrufen zum Blocken der IP Adresse kommen kann.
Das macht mich unsicher und deshalb hab ich mir den Stick besorgt!

Ich sehe im HomeAssistent bisher nur die Geräte (Schaltsteckdose 1+2), die ich im virtuellen RaspberryMatic angemeldet habe.
Diese musste ich in der originalen Homematic IP CCU löschen, sie auf Werkseinstellungen zurücksetzen und in der virtuellen RaspberryMatic anmelden.
Obwohl ich auch die originale HmIP CCU sehe, sehe ich die Geräte, die auf der originalen CCU angemeldet sind nicht.

Was denkst du, was ist da der richtige Weg, den ich gehen sollte?

Vielen Dank vorab für deine Hilfe!

**Comment by Baxxy13** [comment]
> Kann ich die originale CCU weiter laufen lassen

Ja

> diese sowohl aus Home Assistent

Ja

> als auch in der Homematic IP APP bedienen

Eine CCU / RaspberryMatic kann nicht und konnte noch nie über die Homematic-IP App bedient werden. Die App ist einzig für den Access-Point (HAP) nutzbar.

Wenn deine CCU3 funktechnisch besser steht dann nutze sie doch weiter. Du musst halt dann wieder alle Geräte an die CCU3 anlernen.


> Jedoch hab ich gelesen, dass es bei der originalen Zentrale wegen zu vielen Aufrufen zum Blocken der IP Adresse kommen kann.

Wo steht denn sowas? Wenn, dann gilt das für die Integration eines HAP weil die Homeassistant-HmIP-Cloud-Integration "inoffiziell" ist.

Was du letztlich machst musst du selbst entscheiden.

**Reply by dennisxmeyer** [reply]
Danke für deine Antwort!
Ich finde die Geräte der originalen CCU in Homeassistent in der Integration Homematic (IP) nicht.
Unter neue Geräte anlernen, passiert nichts.

In Homeassistent sind beide CCU´s und 2 Schaltsteckdosen.
Ich denke mal das hat mit dem doppelten, jetzt gelöschtem Raspberry zu tun.

![1](https://github.com/jens-maus/RaspberryMatic/assets/157129117/29301130-d210-4df9-97aa-cff1d6eb138f)


Ich kann sie nur in Raspberry neu anlernen auf dem RF-USB Stick.
In Raspberry findet er nur die RF-USB und eine Schaltsteckdose, die originale CCU und die 2. Schaltsteckdose fehlt.

![2](https://github.com/jens-maus/RaspberryMatic/assets/157129117/97af09c1-5475-4ee5-b05c-db8b59c57ea0)


Das ist noch alles schwer verwirrend.

Es wäre gut, wenn ich das Chaos irgendwie noch in den Griff bekomme und weis, wie ich alle Geräte in den Homeassistent bekomme. 
Dann könnte ich mir in HA die Automationen bauen, die ich brauche und HMIP mit Shelly und weiteren kombinieren.

**Reply by MichaelN0815** [reply]
Ich glaube, bei Dir fehlen grundlegende Systemkenntnisse. 

1. Es kann nur einen geben! Wie beim Highlander. Entweder hast Du eine CCU als Chef, bei der alle Geräte angemeldet sind. Und ob da nun die Oroginal-FW oder RM drauf läuft ist dafür auch egal. Oder Du hast eine wie auch immer geartete RM - indem Du die CCU-FW ausgetauscht hast oder eine virtuelle RM betreibst oder als HA-Addon - und daran sind alle deine Geräte angemeldet. Nachdem Du Dich von dem HA-Addon wohl getrennt hast, hast Du jetzt aber immer noch eine RM als VM und eine als echte CCU. Ist immer noch eine zuviel. Von einer musst Du Dich also auch noch trennen. 
2.  Du kannst keine CCU an einer anderen CCU als Gerät anmelden. Siehe 1.

**Reply by dennisxmeyer** [reply]
Nochmals vielen Dank für deine Unterstützung!

Ah ok, also hatte ich quasi 3x Homematic IP am laufen.
Ich hatte verstanden es sind nur 2.

1. HMIP-HAP Access Point für Homematic IP
2. RM als VM
3. RM als Addon in Homeassistent (bereits gelöscht)

**Achtung bitte beachten, ich glaube dass ich keine richtige CCU habe, sondern nur den Hmip-HAP, sprich ein Access Point von Homematic IP.**
**Ergeben sich dadurch andere Voraussetzungen??**

Also VM mit RM löschen mir RM-USB Stick abschalten oder HMIP-HAP abschalten. Hab ich das richtig verstanden?

Wenn ich den originalen HMIP-HAP weiter betreibe, kann ich das auch mit dem Homematic (IP) local Addon machen und das findet dann auch den originalen HMIP-HAP mit allen angelernten Geräten?

**Reply by jens-maus** [reply]
> **Achtung bitte beachten, ich glaube dass ich keine richtige CCU habe, sondern nur den Hmip-HAP, sprich ein Access Point von Homematic IP.** **Ergeben sich dadurch andere Voraussetzungen??**

Ohweh. Ganz ohne dir nahe treten zu wollen, aber du hast da schon nen ganz schönes durcheinander am laufen bzw. hast hier ziemlich viel Aktionismus an den Tag gelegt anstatt dich als erstes mal in Ruhe in das Thema ordentlich einzulesen.

Wenn du wirklich aktuell ein `HmIP-HAP` (Schau hinten auf das Typenschild) mit Cloud-Zugang und eQ3-HomematicIP App auf dem Smartphone betreibst dann bist du hier sowas von falsch. Dann musst du dir erst einmal grundlegend überlegen ob du damit zufrieden bist – auch vielleicht weil dir ansonsten das weitere notwendige technische KnowHow oder Motivation fehlt – oder ob du wirklich auf eine CCU-basierte Platform wechseln willst wozu eben auch RaspberryMatic zählt.

Und solltest du dich wirklich für die CCU/RaspberryMatic Variante entscheiden, dann heisst es da etwas mehr Anlauf und noch mehr Planung erst einmal an den Tag zu legen, denn auch hier gilt wieder das Motto "Es kann nur einen geben". Also **entweder** hat man seine HomematicIP Geräte am HmIP-HAP mit Cloud  angelernt und nutzt die eQ3 eigene Smartphone APP oder man hat sie an der CCU/RaspberryMatic angelernt und kann dann damit auf der einen Seite viel mehr machen, ist unabhängig von der eQ3 Cloud und damit eigentlich auch besser/lokaler unterwegs, aber auf der anderen Seite muss man dann zwangsläufig auf die eQ3 Smartphone App verzichten (was technologisch nicht wirklich ein verzichten ist, btw.) oder eben andere Smartphone Apps wie "pocketControl" oder "tinyMatic" oder eben die HomeAssistant eigene App nutzen um seine Gerätschaften zu steuern, usw.

Du solltest dich also IMHO erst einmal grundlegender in das Ganze Thema "Homematic" und HAP/Cloud vs. CCU/RaspberryMatic/NoCloud einlesen, dir ein entsprechend umfangreiches Wissen aneignen und dann zur Tat schreiten wenn du einen Plan hast. Oder aber du bleibt bei dem was du hast weil du vielleicht ohnehin nicht die Zeit, Lust oder technisches KnowHow hast um mittel- oder längerfristig hier dich mit dem Thema tiefer zu beschäftigen...

**Reply by dennisxmeyer** [reply]
Ich habe mich total verwirren lassen, dass es keine echte CCU sondern nur der HmIP-HAP mit Cloud-Zugang ist.

Ich habe mich bestimmt schon über 100 Stunden damit beschäftigt, aber ich denke es ist nach dem Motto, wer viel liest bzw. Videos schaut, bekommt auch viele Antworten gelaufen.
Das hat mich völlig verwirrt.

Da ich das System Synology/VM RaspberryMatic mit RF-USB/ VM Homeassistent jetzt schon laufen habe, kann ich das ja auch weiter nutzen.

Mein Grundziel ist es, ein System zu finden, welches mir meine Homematic IP Geräte mit anderen Herstellern vereint.
Ich muss auf jeden Fall andere Hersteller mit ins Smarthome einbinden und das auch vom Handy steuern können.

Ich denke ich sollte die Fußboden-Heizungssteuerung erstmal auf dem HMIP-HAP mit der Cloud lassen, um nicht innerhalb eines Wochenendes alles auf ein den Homeassistent umziehen zu müssen.

Mein Gedanke ist es, erstmal alle anderen Geräte (Schalter, Bewegungsmelder etc) über Home Assistent versuchen ein zu binden.
So werde ich auch immer vertrauter mit dem System.

Das Thema der Reichweitenverlängerung, weil der RF-USB Stick in der letzten Ecke des Hauses steht, hat mich mit aller Macht versuchen lassen, das System über den HMIP-HAP laufen zu lassen. Wo nur die Verwirrung zwischen HMIP-HAP und CCU auf kam! 
Die Funktion der Reichweitenverlängerung über die Schaltsteckdosen habe ich im Homeassistent und in RaspberryMatic noch nicht finden können. Das macht mir noch derbe Probleme

**Reply by jens-maus** [reply]
> Ich denke ich sollte die Fußboden-Heizungssteuerung erstmal auf dem HMIP-HAP mit der Cloud lassen, um nicht innerhalb eines Wochenendes alles auf ein den Homeassistent umziehen zu müssen.
> 
> Mein Gedanke ist es, erstmal alle anderen Geräte (Schalter, Bewegungsmelder etc) über Home Assistent versuchen ein zu binden. So werde ich auch immer vertrauter mit dem System.

Das klingt schon mal nach einem vernünftigeren Plan. Ich würde vmtl. so vorgehen das ich sukzessive vom HAP zu RaspberryMatic/CCU umsteige wenn man das nicht auf einmal machen will/kann. D.h. einzelne Geräte nun vom HAP bzw. der Cloud abmelden, werksresetten und dann an der RaspberryMatic/CCU anlernen um dann über die "Homematic(IP) Local" Integration in HomeAssistant aufzutauchen und dort nutzbar zu sein. Und dann geht man so Stück für Stück weiter vor bis kein Gerät mehr im HAP bzw. in der eq3-Cloud angelernt ist.
 
> Das Thema der Reichweitenverlängerung, weil der RF-USB Stick in der letzten Ecke des Hauses steht, hat mich mit aller Macht versuchen lassen, das System über den HMIP-HAP laufen zu lassen. Wo nur die Verwirrung zwischen HMIP-HAP und CCU auf kam! 

Du solltest auch im Blick behalten, das du – wenn du den HmIP-HAP nicht mehr als Cloud-Zugangs-Router nutzt weil du alle Geräte dann in der CCU/RaspberryMatic hast – den HmIP-HAP mit einer anderen Firmware flashen lassen kannst, sodass dieser dann eben nicht mehr als Cloud-Zugangs-Router agiert, sondern faktisch als Funk-Reichweitenverlängerungs-Hub für die homematicIP Geräte die in der CCU/RaspberryMatic angelernt sind. Konkret bedeutet das, das man einen HmIP-HAP (wenn er nicht als Cloud-Gateway verwendet wird) direkt in RaspberryMatic/CCU wie ein normales Gerät anlernen lassen kann und dann agiert er quasi als Reichweitenverlängerungs-Router denn du dann beliebig in deinem Gebäude verteilen kannst um die insgesamte Reichweite zu vergrößern. Und wenn das dann immer noch nicht reicht oder du es gerne ohne einen HAP die Funkreichweite verlängern willst, solltest du darüber nachdenken einfach eine externe Antenne an deinen HmIP-RFUSB dranzulöten. Das kann mitunter wunder bewirken.

**Reply by dennisxmeyer** [reply]
Vielen Dank für deine Mühe! 
Aber vorhandene Schaltsteckdosen als Funk-Reichweitenverlängerung zu nutzen, zumindest bis ich alle Geräte vom HmIP-HAP runter habe ist nicht möglich oder?

**Reply by jens-maus** [reply]
Doch, das ist möglich. Da gibt es aber nichts wirklich einzustellen oder "zu nutzen". Du kannst diese `HmIP-PSM` bzw. `HmIP-PS` natürlich versuchen in deinem Gebäude zu verteilen in der Hoffnung das diese dann die Funksignale weiterreichen. Hier gibt es aber wie gesagt keine Möglichkeit irgendetwas einzustellen oder gar (wie beim alten HomeMatic Protokoll) zu definieren welche Geräte über welche Steckdose "geroutet" werden soll. Das macht das Funknetzwerk im Grunde vollkommen autark und selbstständig und leider auch ohne Einblick oder Kontrolle beim Endnutzer. Erfahrungsgemäß bringt es aber mehr eine externe Antenne zu nutzen oder im zweifel dann eben solche HAPs als Funkreichweitenverlängerer an neuralgischen Stellen einzusetzen. 

Und der Gold-Standard bleibt natürlich: Strippen/Kabel ziehen wo immer das geht und am besten ganz auf Funkkommunikation verzichten.

**Reply by dennisxmeyer** [reply]
Ich habe die HmIP-PS  jetzt quer durch die gesamte Wohnung in jedem Raum eine, bis ans andere Ende ins Wohnzimmer verteilt.
Leider ist der Empfang des letzten HmIP-PS immer noch nicht besser geworden.

Dann werde ich jetzt versuchen erstmal alle Geräte um zu ziehen, die am weitesten weg sind.
Bis ich dann alle in RaspberryMatic umziehen kann und den HmIP-HAP als Verlängerung nutzen kann.

Es fehlten mir wirklich noch Grundlagen, ich habe mir dein Video vom Homematic User Treffen 2019 angesehen und es wird mir in Verbindung mit: ich habe nur den HmIP-HAP vieles deutlich klarer! 

Habe ich folgendes jetzt richtig verstanden?

Ich nutze jetzt:

Hardware:                               
Statt z.B. einer CCU3 --> die Synology NAS, mit RF-USB Stick (über den Virtual Maschine Manager, alternativ wäre auch ein Raspberry Pi möglich gewesen)

Software für die HM / HMIP bzw. bei mir nur HMIP Geräte:     
statt der original eQ3 Software (auf der CCU3) --> Raspberry Matic (über den Virtual Maschine Manager)

Somit hab ich mein bisheriges System über den HMIP-HAP in der Cloud abgelöst und es läuft auf meinem lokalen Synology NAS mit Zugriff über die WebUI.  Die Datensicherung muss ich selbst durchführen.

Um HM bzw. HMIP Geräte mit weiteren Herstellen verknüpfen zu können, habe ich HomeAssistent als Software gewählt. 
HomeAssistent ist auf dem Synology NAS, als extra Maschine im Virtual Maschine Manager installiert.
Auch hiervon muss ich die Datensicherung selbst durchführen.

Durch die Homematic(IP) local Integration und ggf. weitere Integrationen von Geräten anderer Hersteller im HomeAssistent, kann ich somit alle Geräte in Homeassistent zusammenführen und erstelle dort auch alle Steuerungen und Automationen für mein gesamtes Smarthome. Richtig??

Das heisst, ich melde die Geräte in meinem lokalen RaspberryMatic nur an und diese werden dann zu Homeassistent weitergereicht, wo alle weiteren Steuerungen stattfinden?
Oder schlägst du vor, vorher schon weitere Einstellungen für die Geräte in RaspberryMatic durchzuführen?

Zum guten Schluss, muss ich ja auch per Handy auf mein Smarthome zugreifen können. 
Das mache ich dann z.B. mit der Nabu Casa Cloud, Alternativen kenne ich nicht.

Um von außen auf die NAS zu kommen, benutze ich Synology QuickConnect.

Um von außen auf die Weboberfläche meines Raspberry Matic und meine restlichen Netzwerkgeräte zugreifen zu können, erstelle ich auf meiner Unifi Dreammachine mit Draytek Vigor 167 Modem einen VPN über WireGuard oder richte einen DynDns ein.
Was denkst du ist da der bessere Weg?


Vielen vielen Dank für deine wirklich große Mühe, mir zu helfen das Chaos im meinem Kopf zu beseitigen!
